qiskit_optimization/converters/linear_inequality_to_penalty.py

```diff
@@ -0,0 +1,319 @@
+                        )
+
+        # Copy quadratic_constraints
+        for constraint in self._src.quadratic_constraints:
+            self._dst.quadratic_constraints.append(constraint)
+
+        if self._src.objective.sense == QuadraticObjective.Sense.MINIMIZE:
+            self._dst.minimize(offset, linear, quadratic)
+        else:
+            self._dst.maximize(offset, linear, quadratic)
+
+        # Update the penalty to the one just used
+        self._penalty = penalty  # type: float
+
+        return self._dst
+
+    def _conversion_matrix(self, constraint) -> np.ndarray:
+        """Construct conversion matrix for special constraint.
+
+        Returns:
+            Return conversion matrix which is used to construct
+            penalty term in main function.
+
+        """
+        vars_dict = constraint.linear.to_dict(use_name=True)
+        vrs = list(vars_dict.items())
+        rhs = constraint.rhs
+        sense = constraint.sense
+
+        num_vars = len(vrs)
+        combinations = list(itertools.combinations(np.arange(num_vars), 2))
+
+        # conversion matrix
+        conv_matrix = np.zeros((num_vars + 1, num_vars + 1), dtype=int)
+
+        for combination in combinations:
+            index1 = combination[0] + 1
+            index2 = combination[1] + 1
+
+            if rhs == 1:
+                conv_matrix[0][0] = 1 if sense != ConstraintSense.LE else 0
+                conv_matrix[0][index1] = -1 if sense != ConstraintSense.LE else 0
+                conv_matrix[0][index2] = -1 if sense != ConstraintSense.LE else 0
+                conv_matrix[index1][index2] = 1
+            elif rhs == 0:
+                conv_matrix[0][0] = 0
+                if vrs[index1 - 1][1] > 0.0:
+                    conv_matrix[0][index1] = 1
+                elif vrs[index2 - 1][1] > 0.0:
+                    conv_matrix[0][index2] = 1
+                conv_matrix[index1][index2] = -1
+
+        return conv_matrix
+
+    def _is_special_constraint(self, constraint) -> bool:
+        """Determine if constraint is special or not.
+
+        Returns:
+            True: when constraint is special
+            False: when constraint is not special
+        """
+        params = constraint.linear.to_dict()
+        rhs = constraint.rhs
+        sense = constraint.sense
+        coeff_array = np.array(list(params.values()))
+
+        # Binary parameter?
+        if not all(self._src.variables[i].vartype == Variable.Type.BINARY for i in params.keys()):
+            return False
+
+        if len(params) == 2:
+            if rhs == 1:
+                if all(i == 1 for i in params.values()):
+                    if sense in (Constraint.Sense.LE, Constraint.Sense.GE):
+                        # x+y<=1
+                        # x+y>=1
+                        return True
+            if rhs == 0:
+                if sense == Constraint.Sense.LE:
```

Comment:
Could you handle `sense == GE` as well as `LE`?

Reply:
Since GE case as well as  x_1,...,x_n, i.e., \sum_i^n x_i <= 1 => P (\sum_{i, j : i < j} x_i x_j), we can extend the features. Considering that this is part of Qiskit mentor program (which ends on 2021/Jun/10th). We have two option.

1.  Reject or Approve this PR to finish mentoring program, and raise new feature enhancement.
2. Continue to use this PR to implement those.

We are willing to extent the capability, and can contribute beyond Qiskit mentoring program. Because the purpose of mentoring is to become Qiskit contributor, not stop working.

Reply:
You have already published this PR. This is good enough to report as the result of the mentor program even if it is not merged by June 10.
Please continue revising the PR. The generalization would be beneficial.